### PR TITLE
imxrt-multi: uart parity bit appending fix

### PR DESCRIPTION
**Pull request by gorbanchovie**
JIRA: NIL-287

## Description

A fix to appending parity bit to UART data frames for i.MX RT MCUs.
## Motivation and Context

UART configuration for i.MX RT MCUs is different then, for example, for i.MX 6ULL. Parity bit is treated as a part of data bits. So in order to append parity bit to data frame, data length must be incremented.
## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## How Has This Been Tested?

- [ ] Already covered by automatic testing.
- [ ] New test added: (add PR link here).
- [x] Tested by hand on: i.MX RT1170.

## Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing linter checks and tests passed.
- [x] My changes generate no new compilation warnings for any of the targets.

## Special treatment

- [ ] This PR needs additional PRs to work (list the PRs, preferably in merge-order).
- [ ] I will merge this PR by myself when appropriate.

### CODE REVIEW

**Review by agkaminski — DISMISSED** [review]
LGTM, cross-checked with the RM

File: multi/imxrt-multi/uart.c [hunk 1]
```diff
@@ -155,24 +155,28 @@ static void signal_txready(void *_uart)
...
+
+	if ((*cflag & CSIZE) == CS7 && !(*cflag & PARENB)) {
+		t |= 1 << 11;
 	}
 	else { /* CS8 */
+		if (*cflag & PARENB) {
+			t |= 1 << 4;
+		}
 		*cflag &= ~CSIZE;
 		*cflag |= CS8;
```
Comment (anglov) [inline multi/imxrt-multi/uart.c:177]: It looks invalid.  When you have eg. 7E1  it will set CSIZE to CS8

**Review by anglov — CHANGES_REQUESTED** [review]

**Review by anglov — APPROVED** [review]
Solution looks hacky but I think it's valid

**Review by agkaminski — APPROVED** [review]
@anglov  Why hacky? One thing I see (but it's present in the old code too) is that we silently set CS8 if CSIZE != CS7.

**Pull request merged**